### PR TITLE
Adding Redis Exporter

### PR DESCRIPTION
**Pull request by sfawcett123**
### [Add Redis Exported ](https://trello.com/c/S9rH1dAV/226-add-redis-exporter-to-prometheusall)

Added the redis exported to Prometheus ALL, also at the same time added some missing variables, which allows GiT to convert to prometheus_all

### Testing
https://prometheus-sb-get-into-teaching.london.cloudapps.digital/targets
https://grafana-sb-get-into-teaching.london.cloudapps.digital/?orgId=1
https://alertmanager-sb-get-into-teaching.london.cloudapps.digital/#/alerts

### CODE REVIEW

File: prometheus_all/input.tf [hunk 1]
```diff
@@ -15,12 +15,17 @@ variable grafana_google_client_secret { default = "" }
...
+variable grafana_google_jwt { default = "" }
+variable grafana_runtime_version { default = ""} 
 
 variable prometheus_memory { default = null }
 variable prometheus_disk_quota { default = null }
+variable prometheus_extra_scrape_config { default = "" }
 
 variable influxdb_service_plan { default = "tiny-1_x" }
 
+variable redis_service_instance_id { default = null }
```
Comment (saliceti) [inline prometheus_all/input.tf:27]: Can we pass a name instead? And add a data source to resolve the name to the id

Comment (sfawcett123) [reply prometheus_all/input.tf:27]: In my case where monitoring is  in a different space to the Redis Service, you would need to pass both the Name of the Redis Instance and the Name of the Space, then resolve both of them.

File: prometheus_all/resources.tf [hunk 2]
```diff
@@ -8,6 +8,14 @@ module paas_prometheus_exporter {
   paas_password            = var.paas_exporter_password
 }
 
+module "redis_prometheus_exporter" {
+  source                    = "../redis_prometheus_exporter"
+  count                     = var.redis_service_instance_id == "" ?  0 : 1
```
Comment (saliceti) [inline prometheus_all/resources.tf:13]: Shouldn't it be `var.redis_service_instance_id == "" ?`

File: prometheus_all/resources.tf [hunk 3]
```diff
@@ -24,11 +32,13 @@ module prometheus {
   monitoring_instance_name          = var.monitoring_instance_name
   monitoring_space_id               = data.cloudfoundry_space.monitoring.id
   paas_prometheus_exporter_endpoint = module.paas_prometheus_exporter[0].endpoint
+  redis_prometheus_exporter_endpoint= var.redis_service_instance_id == "" ? "" : module.redis_prometheus_exporter[0].endpoint
```
Comment (saliceti) [inline prometheus_all/resources.tf:35]: Shouldn't it be
`var.redis_service_instance_id == null ? null : module.redis_prometheus_exporter[0].endpoint`

Comment (sfawcett123) [reply prometheus_all/resources.tf]: agreed

**Review by saliceti — APPROVED** [review]

File: prometheus_all/input.tf [hunk 4]
```diff
@@ -15,12 +15,17 @@ variable grafana_google_client_secret { default = "" }
...
+variable grafana_google_jwt { default = "" }
+variable grafana_runtime_version { default = "" }
 
 variable prometheus_memory { default = null }
 variable prometheus_disk_quota { default = null }
+variable prometheus_extra_scrape_config { default = "" }
 
 variable influxdb_service_plan { default = "tiny-1_x" }
 
+variable redis_service_instance_id { default = null }
```
Comment (saliceti) [inline prometheus_all/input.tf:27]: Can we pass a list instead? We may have multiple redis instances to monitor (in teaching vacancies for example)

**Comment by saliceti** [comment]
[Add redis exporter to prometheus_all](https://trello.com/c/S9rH1dAV/226-add-redis-exporter-to-prometheusall)

**Pull request merged**